### DISCUSSION TITLE
How to get the realtime validation state?

### DISCUSSION THREAD

**Discussion by richerfu** [react-hook-form/react-hook-form#6043] [Q&A]
Now, i need to get the realtime validation state in order to disabled the submit button.
i try to use `trigger()`,but `formState.errors` will rerender component and render all errors.

just like this.

https://codesandbox.io/s/strange-dhawan-kk5vp?file=/src/App.tsx

what i should do?

**Comment by writes-of-spring** [comment]
Are you tied to using v6?

With version 6, I would add validation mode to the useForm hook, and use (by destructuring) isValid and isDirty from formState.

then the button is disabled with it's !isValid || !isDirty.

Example here: 

https://codesandbox.io/s/upbeat-mclean-oewv6?file=/src/App.tsx

**Reply by richerfu** [reply]
@PeterKerins  But i must use `onSubmit` mode for some reason.And `isValid` || `isDirty` always get false or wrong value .

https://codesandbox.io/s/sleepy-pasteur-cpqxc?file=/src/App.tsx

**Comment by bluebill1049** [comment]
why not subscribe to isValid formState?

**Reply by richerfu** [reply]
like this demo.
https://github.com/react-hook-form/react-hook-form/discussions/6043#discussioncomment-1037489

https://codesandbox.io/s/sleepy-pasteur-cpqxc?file=/src/App.tsx

`isValid` is always false.

**Reply by bluebill1049** [reply]
<img width="857" alt="Screen Shot 2021-07-23 at 1 00 28 pm" src="https://user-images.githubusercontent.com/10513364/126732826-6fd6e5b5-7740-47f5-903c-45f7835488fe.png">

**Reply by richerfu** [reply]
yeah, i must use `onSubmit` mode for some reason. So `isValid` is not good
